osf/migrations/0177_update_schema_data.py

```diff
@@ -0,0 +1,285 @@
+    ('textarea', 'osf-author-import'): 'contributors-input',
+    ('textarea', 'string'): 'long-text-input',
+    ('textarea-lg', None): 'long-text-input',
+    ('textarea-lg', 'string'): 'long-text-input',
+    ('textarea-xl', 'string'): 'long-text-input',
+}
+
+def remove_version_1_schemas(state, schema):
+    RegistrationSchema = state.get_model('osf', 'registrationschema')
+    assert RegistrationSchema.objects.filter(schema_version=1, abstractnode__isnull=False).count() == 0
+    assert RegistrationSchema.objects.filter(schema_version=1, draftregistration__isnull=False).count() == 0
+    RegistrationSchema.objects.filter(schema_version=1).delete()
+
+def update_schemaless_registrations(state, schema):
+    RegistrationSchema = state.get_model('osf', 'registrationschema')
+    AbstractNode = state.get_model('osf', 'abstractnode')
+
+    open_ended_schema = RegistrationSchema.objects.get(name='Open-Ended Registration')
+    open_ended_meta = {
+        '{}'.format(open_ended_schema._id): {
+            'summary': {
+                'comments': [],
+                'extra': [],
+                'value': ''
+            }
+        }
+    }
+
+    schemaless_regs_with_meta = AbstractNode.objects.filter(type='osf.registration', registered_schema__isnull=True).exclude(registered_meta={})
+    schemaless_regs_without_meta = AbstractNode.objects.filter(type='osf.registration', registered_schema__isnull=True, registered_meta={})
+
+    for reg in schemaless_regs_without_meta.all():
+        reg.registered_schema.add(open_ended_schema)
+        reg.registered_meta = open_ended_meta
+        reg.save()
+
+    for reg in schemaless_regs_with_meta.all():
+        reg.registered_schema.add(RegistrationSchema.objects.get(_id=reg.registered_meta.keys()[0]))
+
+def update_schema_configs(state, schema):
+    RegistrationSchema = state.get_model('osf', 'registrationschema')
+    for rs in RegistrationSchema.objects.all():
+        if rs.schema.get('description', False):
+            rs.description = rs.schema['description']
+            rs.save()
+
+def unset_schema_configs(state, schema):
+    RegistrationSchema = state.get_model('osf', 'registrationschema')
+    RegistrationSchema.objects.update(config=dict())
+
+def unmap_formblocks(state, schema):
+    RegistrationFormBlock = state.get_model('osf', 'registrationformblock')
+    RegistrationFormBlock.objects.all().delete()
+
+def noop(*args, **kwargs):
+    pass
+
+def create_block(state, schema_id, block_type, display_text='', required=False, help_text='',
+        answer_id='', chunk_id='', example_text=''):
+    RegistrationFormBlock = state.get_model('osf', 'registrationformblock')
+
+    return RegistrationFormBlock.objects.create(
+        schema_id=schema_id,
+        block_type=block_type,
+        required=required,
+        display_text=display_text,
+        help_text=help_text,
+        answer_id=answer_id,
+        chunk_id=chunk_id,
+        example_text=example_text
+    )
```

Comment:
There are only a few schemas to migrate, depending on the environment <15, and the big schemas will map to less than 200 schema blocks -  so not currently optimizing the save of the related form blocks.